### PR TITLE
Improve Readme

### PR DESCRIPTION
**Pull request by maclobdell**
Updated the readme extensively to add more detail, improve readability, and add a couple images.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,33 +1,79 @@
...
+  - EBS volume size: 10 GiB
+- Configure access and security
+  - Public access: Enabled 
+  - Domain access policy: Allow open access to the domain
+  - Require HTTPS: Enabled
+
+Notes:  
+- Features not specifically mentioned can be disabled. 
+
+- For testing purposes, the smallest cluster setting should be sufficient. Note that this still may not be covered by the AWS free tier, but the cost should be minimal.
```
Comment (marcuschangarm) [inline README.md:66]: Perhaps also mention that the ES version number is not fixed to 7.9, but only that is has been tested with 7.9.

File: README.md [hunk 2]
```diff
@@ -100,21 +146,31 @@ PUT /registrations
...
-* Note the "API Endpoint" URL and the "API key".
+- In the **Add trigger** template, choose **Create an API** then select **REST API**.
+- Under *Security*, choose **API key**
+- Enable logging. Note, this may fail due to missing permissions. Follow the guide above about CloudWatch roles and try again.
+- Note the **API Endpoint** URL and the **API key**.
 
 ### Pelion Cloud REST API
 
-From the Arm Pelion Device Management portal, create a new API key. From a command terminal execute:
+- Log in to the [Pelion Device Management portal](https://portal.mbedcloud.com/login).
```
Comment (bentcooke) [inline README.md:175]: Maybe add a bit of an intro here as to why these REST call are necessary.  Something like:

Now that your AWS account is configured to receive the notification messages from Pelion, the call back URL must be set on your Pelion account and subscriptions set to start the message flow.

File: README.md [hunk 3]
```diff
@@ -126,9 +182,14 @@ curl -X PUT https://api.us-east-1.mbedcloud.com/v2/notification/callback \
 }'
 ```
 
-Where `PELION_API_KEY` is the API key from Pelion Device Management portal, `AWS_API_ENDPOINT` and `AWS_API_KEY` are the API endpoint and key retrieved from AWS API Gateway above. 
+Where `PELION_API_KEY` is the access key from Pelion Device Management portal, `AWS_API_ENDPOINT` and `AWS_API_KEY` are the API endpoint and key retrieved from AWS API Gateway earlier. 
+
+If all goes well, you should get a **204** response back.
+
+- Next, enable pre-subscriptions for LwM2M resources that you want to be stored in Elasticsearch:
```
Comment (bentcooke) [inline README.md]: Subscripts and presubscription can be managed with the ['v2/subscriptions'](https://developer.pelion.com/docs/device-management-api/connect/#/Device%20data%20-%20subscriptions) REST API endpoint on Pelion.

**Review by bentcooke — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (maclobdell) [reply README.md:66]: fixed in 4b755d7

File: README.md (hunk 2 above)
Comment (maclobdell) [reply README.md:175]: good idea. added more context in 4b755d7.

File: README.md (hunk 3 above)
Comment (maclobdell) [reply README.md]: added in 4b755d7

File: README.md [hunk 4]
```diff
@@ -100,21 +146,35 @@ PUT /registrations
...
-* Note the "API Endpoint" URL and the "API key".
+- In the **Add trigger** template, choose **Create an API** then select **REST API**.
+- Under *Security*, choose **API key**
+- Enable logging. Note, this may fail due to missing permissions. Follow the guide above about CloudWatch roles and try again.
+- Note the **API Endpoint** URL and the **API key**.
 
 ### Pelion Cloud REST API
 
-From the Arm Pelion Device Management portal, create a new API key. From a command terminal execute:
+Now that your AWS account is configured to receive notification messages, you must configure Pelion to generate and send the messages to the right place. This is done by setting up a callback URL and subscribing to specific resources in your Pelion account. Once this is completed, the messages will start to flow.
```
Comment (marcuschangarm) [inline README.md:171]: 4sncg7

File: README.md [hunk 5]
```diff
@@ -100,21 +146,35 @@ PUT /registrations
...
+- Search for and open the **Lambda** dashboard.
+- Choose **Create Function** and **Author From Scratch**. 
+- Pick a suitable function name (e.g. pass_pelion_data)
+- Choose at least Node.js 12.x 
+- Select *Create a new role with basic Lambda permissions* or *Use an existing role* if you already have one. 
+- Click **Create Function**
+- Copy the contents of `index.js` from this repository into the file named `index.js` within the Lambda Function code workspace.
+- Create a new file in the Lambda workspace named `elasticsearch.js` and copy the contents from the file `elasticsearch.js` from this repository into it. 
+- Under *Runtime settings*, ensure the field **Handler** is set to **index.handler**.
+- Under **Environment Variables**, create a new variable `ELASTICSEARCH_DOMAIN` with the Elasticsearch *Endpoint* name from earlier without the "https://".
```
Comment (maclobdell) [inline README.md:158]: @marcuschangarm does this work for you?

Comment (maclobdell) [reply README.md:158]: based on our slack conversation, the answer is yes.

Comment (marcuschangarm) [reply README.md:158]: yes

**Review by bentcooke — APPROVED** [review]
Looks good...  Thanks Mac!

**Pull request merged**